### PR TITLE
Use versioned filenames for tls wrapper and quote instances

### PR DESCRIPTION
**Pull request by HuangYijuan**
Fixes: #701

### CODE REVIEW

File: rats-tls/src/attesters/internal/etls_enclave_attester_load_single.c [hunk 1]
```diff
@@ -16,7 +16,7 @@
 #ifdef SGX
 #define PATTERN_SUFFIX ".a"
 #else
-#define PATTERN_SUFFIX ".so"
+#define PATTERN_SUFFIX ".so.0.1.0"
```
Comment (YangLiang3) [inline rats-tls/src/attesters/internal/etls_enclave_attester_load_single.c:19]: library SUFFIX  shoud not be fixed

File: rats-tls/src/crypto_wrappers/internal/etls_crypto_wrapper_load_single.c [hunk 2]
```diff
@@ -16,7 +16,7 @@
 #ifdef SGX
 #define PATTERN_SUFFIX ".a"
 #else
-#define PATTERN_SUFFIX ".so"
+#define PATTERN_SUFFIX ".so.0.1.0"
```
Comment (YangLiang3) [inline rats-tls/src/crypto_wrappers/internal/etls_crypto_wrapper_load_single.c:19]: ditto.

File: rats-tls/src/tls_wrappers/internal/etls_tls_wrapper_load_single.c [hunk 3]
```diff
@@ -16,7 +16,7 @@
 #ifdef SGX
 #define PATTERN_SUFFIX ".a"
 #else
-#define PATTERN_SUFFIX ".so"
+#define PATTERN_SUFFIX ".so.0.1.0"
```
Comment (YangLiang3) [inline rats-tls/src/tls_wrappers/internal/etls_tls_wrapper_load_single.c:19]: ditto.

File: rats-tls/src/verifiers/internal/etls_enclave_verifier_load_single.c [hunk 4]
```diff
@@ -16,7 +16,7 @@
 #ifdef SGX
 #define PATTERN_SUFFIX ".a"
 #else
-#define PATTERN_SUFFIX ".so"
+#define PATTERN_SUFFIX ".so.0.1.0"
```
Comment (YangLiang3) [inline rats-tls/src/verifiers/internal/etls_enclave_verifier_load_single.c:19]: ditto.

File: rats-tls/src/attesters/internal/etls_enclave_attester_load_all.c [hunk 5]
```diff
@@ -43,13 +43,15 @@ enclave_tls_err_t etls_enclave_attester_load_all(void)
 	while ((etls_readdir(dir, &ptr)) != 1) {
 		if (!strcmp(ptr->d_name, ".") || !strcmp(ptr->d_name, ".."))
 			continue;
+		if (strncmp(ptr->d_name+strlen(ptr->d_name)-2, "so", 2))
```
Comment (jiazhang0) [inline rats-tls/src/attesters/internal/etls_enclave_attester_load_all.c:46]: Change to
```C
strncmp(ptr->d_name + strlen(ptr->d_name) - 3, ".so", 3)
```

File: rats-tls/src/attesters/internal/etls_enclave_attester_load_all.c [hunk 6]
```diff
@@ -43,13 +43,15 @@ enclave_tls_err_t etls_enclave_attester_load_all(void)
...
+			continue;
 
 #ifdef OCCLUM
 		/* Occlum can't identify the d_type of the file, always return DT_UNKNOWN */
 		if (strncmp(ptr->d_name + strlen(ptr->d_name) - strlen(PATTERN_SUFFIX),
 			    PATTERN_SUFFIX, strlen(PATTERN_SUFFIX)) == 0) {
 #else
-		if (ptr->d_type == DT_REG) {
+		if (ptr->d_type == DT_REG || ptr->d_type == DT_LNK) {
```
Comment (jiazhang0) [inline rats-tls/src/attesters/internal/etls_enclave_attester_load_all.c:54]: Why checking symbol link type?

File: rats-tls/src/crypto_wrappers/internal/etls_crypto_wrapper_load_all.c [hunk 7]
```diff
@@ -47,12 +47,15 @@ enclave_tls_err_t etls_crypto_wrapper_load_all(void)
 		if (!strcmp(ptr->d_name, ".") || !strcmp(ptr->d_name, "..")) {
 			continue;
 		}
+		if (strncmp(ptr->d_name+strlen(ptr->d_name)-2, "so", 2)) {
```
Comment (jiazhang0) [inline rats-tls/src/crypto_wrappers/internal/etls_crypto_wrapper_load_all.c:50]: ditto

File: rats-tls/src/crypto_wrappers/internal/etls_crypto_wrapper_load_all.c [hunk 8]
```diff
@@ -47,12 +47,15 @@ enclave_tls_err_t etls_crypto_wrapper_load_all(void)
...
+			continue;
+		}
 #ifdef OCCLUM
 		/* Occlum can't identify the d_type of the file, always return DT_UNKNOWN */
 		if (strncmp(ptr->d_name + strlen(ptr->d_name) - strlen(PATTERN_SUFFIX),
 			    PATTERN_SUFFIX, strlen(PATTERN_SUFFIX)) == 0) {
 #else
-		if (ptr->d_type == DT_REG) {
+		if (ptr->d_type == DT_REG || ptr->d_type == DT_LNK) {
```
Comment (jiazhang0) [inline rats-tls/src/crypto_wrappers/internal/etls_crypto_wrapper_load_all.c:58]: ditto

**Review by jiazhang0 — CHANGES_REQUESTED** [review]

File: rats-tls/src/attesters/internal/etls_enclave_attester_load_all.c (hunk 6 above)
Comment (YangLiang3) [reply rats-tls/src/attesters/internal/etls_enclave_attester_load_all.c:54]: In libs dir, there are soft link of  instance libs which can be quoted by application.

**Review by YangLiang3 — APPROVED** [review]
LGTM

**Review by jiazhang0 — APPROVED** [review]

**Pull request merged**